tests/MongoDB.Driver.Tests/AuthenticationTests.cs

```diff
@@ -294,27 +297,71 @@ public void Authentication_succeeds_when_user_has_multiple_credentials_and_mecha
         }
 #endif
-        
-        private void AssertAuthenticationSucceeds(MongoClientSettings settings)
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Authentication_succeeds_with_MONGODB_X509_mechanism(
+            [Values(false, true)] bool async)
         {
-             new MongoClient(settings).ListDatabaseNames().ToEnumerable().ToList();
+            RequireEnvironment.Check().EnvironmentVariable("MONGO_X509_CLIENT_CERTIFICATE_PATH", isDefined: true);
+            RequireEnvironment.Check().EnvironmentVariable("MONGO_X509_CLIENT_CERTIFICATE_PASSWORD", isDefined: true);
+            RequireServer.Check().Tls(required: true);
+
+            var pathToClientCertificate = Environment.GetEnvironmentVariable("MONGO_X509_CLIENT_CERTIFICATE_PATH");
+            var password = Environment.GetEnvironmentVariable("MONGO_X509_CLIENT_CERTIFICATE_PASSWORD");
+            var clientCertificate = new X509Certificate2(pathToClientCertificate, password);
+
+            var userName = GetRfc2253FormattedUsernameFromX509ClientCertificate(clientCertificate);
+            DropDatabaseUser(DriverTestConfiguration.Client, database: "$external", userName);
+            CreateX509DatabaseUser(DriverTestConfiguration.Client, userName);
+
+            var settings = DriverTestConfiguration.GetClientSettings().Clone();
+            var serverVersion = CoreTestConfiguration.ServerVersion;
+            var serverExtractsUsernameFromCertificate =
+                serverVersion >= Feature.ServerExtractsUsernameFromX509Certificate.FirstSupportedVersion;
+            settings.Credential = MongoCredential.CreateMongoX509Credential(
+                username: serverExtractsUsernameFromCertificate ? null : userName);
+            settings.SslSettings = settings.SslSettings.Clone();
+            settings.SslSettings.ClientCertificates = new[] { clientCertificate };
+
+            AssertAuthenticationSucceeds(settings, async);
+        }
+
+        private void AssertAuthenticationSucceeds(MongoClientSettings settings, bool async = false)
+        {
+            /* If we don't use a DisposableClient, the second run of AuthenticationSucceedsWithMongoDB_X509_mechanism
+             * will fail because the backing Cluster's connections will be associated with a dropped user */
+            using (var client = DriverTestConfiguration.CreateDisposableclient(settings))
+            {
+                /* The first command executed with the MongoClient triggers either the sync or async variation of the
+                 * MongoClient's IAuthenticator */
+                if (async)
+                {
+                    var cursor = client.ListDatabaseNamesAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+                    _ = cursor.ToEnumerable().ToList();
+                }
+                else
+                {
+                    _ = client.ListDatabaseNames().ToEnumerable().ToList();
```

Comment:
nit: calling `ToEnumerable` is not necessary (and actually an anti-pattern). We can just use:

    _ = cursor.ToList();

and

    _ = client.ListDatabaseNames().ToList();

Note: I realize of course that the `ToEnumerable` was already there...

Reply:
Fixed!